src/hardware/mixer.cpp

```diff
@@ -769,16 +828,16 @@ void MixerChannel::AddStretched(uint16_t len, int16_t *data)
 {
 	MIXER_LockAudioDevice();
 
-	if (done >= needed) {
+	if (frames_done >= frames_needed) {
 		LOG_MSG("Can't add, buffer full");
 		MIXER_UnlockAudioDevice();
 		return;
 	}
 	//Target samples this inputs gets stretched into
-	auto outlen = needed - done;
+	auto outlen = frames_needed - frames_done;
 	auto index = 0;
 	auto index_add = (len << FREQ_SHIFT) / outlen;
-	auto mixpos = check_cast<work_index_t>(mixer.pos + done);
+	auto mixpos = check_cast<work_index_t>(mixer.pos + frames_done);
```

Comment:
Should we have the same `& MIXER_BUFMASK` wrap-around behavior here too?

Reply:
It's there in the middle of the loop a few lines below.